### PR TITLE
help

### PR DESCRIPTION
**Pull request by pyapyapya**
@kozistr

### CODE REVIEW

File: feature_extract_model.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
+"""
+질문할 것
+1. DataSet을 만들었습니다. DataLoader도 만들어야 하나요?
```
Comment (kozistr) [inline feature_extract_model.py]: 당연하죠 파이토치 기본 예제들 한번 보셈

File: feature_extract_model.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
+"""
+질문할 것
+1. DataSet을 만들었습니다. DataLoader도 만들어야 하나요?
+2. __getitem__에서 return을 하는 과정에서 [img, label] 이렇게 감싸면 어떻게 되는지 명확하게 알지 못하겠음
```
Comment (kozistr) [inline feature_extract_model.py]: 직접 해 보고 어떻게 리턴되는지, 어떤걸 리턴해야 하는지 봐 보셈

File: feature_extract_model.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
+"""
+질문할 것
+1. DataSet을 만들었습니다. DataLoader도 만들어야 하나요?
+2. __getitem__에서 return을 하는 과정에서 [img, label] 이렇게 감싸면 어떻게 되는지 명확하게 알지 못하겠음
+3. train에서 __getitem__을 호출하면 2 value expect 오류가 발생함 too many하다고. 그런데 len으로 사이즈를 재보면 2개가 나오는데
+어떻게 해결해야할지 잘 모르겠음,, single label과 multi label은 어떤 부분을 다르게 신경써야하는지도 잘 모르겠음
+4. 논문을 최대한 수정하면서 반영하려고 하는데, multi-label에서는 마지막 layer에
```
Comment (kozistr) [inline feature_extract_model.py]: 주로 pretrained 모델을 쓸 때 bottleneck 까지만 쓰고 (last pooling layer) fc layer 는 안씀
즉, 뒤에 마지막 fc layer 는 너가 만들어 붙이면 됨. 한번 pytorch transfer learning 이런거 찾아보셈

File: feature_extract_model.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
+"""
+질문할 것
+1. DataSet을 만들었습니다. DataLoader도 만들어야 하나요?
+2. __getitem__에서 return을 하는 과정에서 [img, label] 이렇게 감싸면 어떻게 되는지 명확하게 알지 못하겠음
+3. train에서 __getitem__을 호출하면 2 value expect 오류가 발생함 too many하다고. 그런데 len으로 사이즈를 재보면 2개가 나오는데
```
Comment (kozistr) [inline feature_extract_model.py]: 이건 직접 디버깅 해 보셈 value expect n too many 는 파이썬 internal error

File: feature_extract_model.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
+"""
+질문할 것
+1. DataSet을 만들었습니다. DataLoader도 만들어야 하나요?
+2. __getitem__에서 return을 하는 과정에서 [img, label] 이렇게 감싸면 어떻게 되는지 명확하게 알지 못하겠음
+3. train에서 __getitem__을 호출하면 2 value expect 오류가 발생함 too many하다고. 그런데 len으로 사이즈를 재보면 2개가 나오는데
+어떻게 해결해야할지 잘 모르겠음,, single label과 multi label은 어떤 부분을 다르게 신경써야하는지도 잘 모르겠음
+4. 논문을 최대한 수정하면서 반영하려고 하는데, multi-label에서는 마지막 layer에
+    softmax가 아닌 sigmoid를 써야한다는데 resnet 모델 구현된 거에서는 model을 softmax 쓸텐데 pre-train된거 따로 구해야하는지??
+5. MSE 방식 썼던데 CrossEntropy써도 상관없는지???
```
Comment (kozistr) [inline feature_extract_model.py]: 선택하기나름 적절한 task 에 적절한 loss 쓰는 게 정답

File: feature_extract_model.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+    epochs = 100
+    batch_size = 16
+    loss_function = CrossEntropyLoss()
+    optimizer = Adam(params=model.parameters(), lr=0.01, betas=(0.9, 0.99), eps=1e-08)
+    lr_scheduler: StepLR = StepLR(optimizer, step_size=20, gamma=0.1)
+
+    for epoch in range(epochs):
+        idx = 0
+        for img, label in dataset.__getitem__(idx):
```
Comment (kozistr) [inline feature_extract_model.py]: dataloader 를 넣어줘야함,. 요렇게 쓰는거 아니다리
pytorch 기본 예제를 먼저 공부해보셈

File: feature_extract_model.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    """
+    model.train()
+    model.cuda()
+    num_data = dataset.__len__()
+
+    epochs = 100
+    batch_size = 16
+    loss_function = CrossEntropyLoss()
+    optimizer = Adam(params=model.parameters(), lr=0.01, betas=(0.9, 0.99), eps=1e-08)
+    lr_scheduler: StepLR = StepLR(optimizer, step_size=20, gamma=0.1)
```
Comment (kozistr) [inline feature_extract_model.py]: 전에 리뷰 해 준거 같은데, StepLR 은 nn.Module 로 hinting 하는게 맞. 같은 이름 쓰는 거면 hinting 에 의미가 없다리

File: feature_extract_model.py [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+        return len(self.txt_file)
+
+    def __getitem__(self, idx):
+        """
+        :param idx: DataSet의
+        :return: img의 tensor, label -> tags를 나타내는 list를 묶어놓은 List를 return
+        """
+        get_image_elements = GetImageElements()
+        img_id: List = get_image_elements.get_img_id()
+        img_tags: List = get_image_elements.get_img_tags()
```
Comment (kozistr) [inline feature_extract_model.py]: 이거 에러나지 않으려나
